pkg/webui/console/components/unit-input/index.js

```diff
@@ -0,0 +1,114 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react'
+import classnames from 'classnames'
+import bind from 'autobind-decorator'
+
+import Select from '../../../components/select'
+import Input from '../../../components/input'
+import { unit as unitRegexp } from '../../lib/regexp'
+import PropTypes from '../../../lib/prop-types'
+import withComputedProps from '../../../lib/components/with-computed-props'
+import style from './unit-input.styl'
+
+@withComputedProps(props => ({
+  ...props,
+  value: props.decode(props.value),
+}))
+class UnitInput extends React.PureComponent {
+  static propTypes = {
+    className: PropTypes.string,
+    decode: PropTypes.func,
+    encode: PropTypes.func,
+    name: PropTypes.string.isRequired,
+    onChange: PropTypes.func.isRequired,
+    units: PropTypes.arrayOf(
+      PropTypes.shape({
+        label: PropTypes.message,
+        value: PropTypes.string,
+      }),
+    ).isRequired,
+    value: PropTypes.shape({
+      unit: PropTypes.string,
+      duration: PropTypes.number,
+    }),
+  }
+
+  static defaultProps = {
+    className: undefined,
+    encode: (duration, unit) => (duration ? `${duration}${unit}` : unit),
+    decode: value => {
+      const duration = value.split(unitRegexp)[0]
+      const unit = value.split(duration)[1]
+      return {
+        duration: duration ? Number(duration) : undefined,
+        unit,
+      }
+    },
+    value: undefined,
+  }
+
+  @bind
+  async handleChange(duration) {
+    const { onChange, encode, value } = this.props
+    // if duration is empty, propagate empty value to form
+    onChange(encode(duration, value.unit))
+  }
+
+  @bind
+  async handleUnitChange(unit) {
+    const { onChange, encode, value } = this.props
+    // if duration is empty, propagate empty value to form
+    onChange(encode(value.duration, unit))
+  }
+
+  render() {
+    const {
+      className,
+      name,
+      units,
+      value: { duration, unit },
+    } = this.props
+
+    const selectTimeUnitComponent = (
+      <Select
+        className={style.select}
+        name={`${name}-select`}
+        options={units}
+        onChange={this.handleUnitChange}
+        value={unit}
+      />
+    )
+
+    return (
+      <React.Fragment>
+        <div className={classnames(className, style.container)}>
+          <Input
+            className={style.number}
+            type="number"
+            step="any"
+            name={name}
+            onBlur={this.handleBlur}
+            value={duration}
```

Comment:
```suggestion
            value={duration}
            required={required}
```
Pass on the required prop to the input component.